ludwig/backend/ray.py

```diff
@@ -683,24 +691,40 @@ def get_resources_per_worker(self) -> Tuple[int, int]:
         num_cpus = resources_per_worker.get("CPU", (1 if num_gpus == 0 else 0))
         return num_cpus, num_gpus
 
-    def batch_predict(self, dataset: RayDataset, *args, collect_logits: bool = False, **kwargs):
+    def batch_predict(
+        self, dataset: RayDataset, *args, collect_logits: bool = False, model_ref: ObjectRef = None, **kwargs
+    ):
         self._check_dataset(dataset)
 
         predictor_kwargs = self.predictor_kwargs
         output_columns = get_output_columns(self.model.output_features, include_logits=collect_logits)
+
+        num_cpus, num_gpus = self.get_resources_per_worker()
+
+        distributed_strategy = self.trainer_kwargs.get("strategy", get_default_strategy_name())
+        if distributed_strategy == "deepspeed" or (
+            isinstance(distributed_strategy, dict) and distributed_strategy.get("type", None) == "deepspeed"
+        ):
+            # make sure all gpus are used by a single Ray Datasets worker during batch predict
+            num_gpus = self.trainer_kwargs.get("num_workers", 1) * num_gpus
```

Comment:
This doesn't quite work, as the Ray cluster could have multiple nodes, but the model can't be loaded across multiple nodes. Therefore, you want to compute the max number of gpus on a single node:

```
num_gpus = max(n["Resources"].get("GPU", 0) for n in ray.nodes())
```

Reply:
got it!